compiler/rustc_ast/src/ast.rs

```diff
@@ -1450,8 +1450,8 @@ pub enum StructRest {
     Base(P<Expr>),
     /// `..`.
     Rest(Span),
-    /// No trailing `..` or expression.
-    None,
+    /// No trailing `..` or expression. The `Span` points at the trailing `,` or spot before `}`.
+    None(Span),
```

Comment:
We can get away with not adding this span if we take the span between the end of the last field and the end of the struct expression (`}`), but that makes things unnecessarily complicated.